Source/settings/mods.ts

```diff
@@ -143,6 +143,53 @@ module FullScreenPokemon {
                         this.ObjectMaker.getFunction("SightDetector").prototype.nocollide = false;
                     }
                 }
+            },
+            {
+                name: "Nuzlocke Challenge",
+                enabled: false,
+                events: {
+                    "onModEnable": function (mod: ModAttachr.IModAttachrMod): void {
+                        return;
+                    },
+                    "onModDisable": function (mod: ModAttachr.IModAttachrMod): void {
+                        return;
+                    },
+                    "onBattleComplete": function (mod: ModAttachr.IModAttachrMod, eventName: string, settings: IBattleInfo): void {
+                        let grass: IGrass = this.player.grass,
+                            grassMap: IMap = grass ? <IMap>this.AreaSpawner.getMap(grass.mapName) : undefined,
+                            grassArea: IArea = grassMap ? <IArea>grassMap.areas[grass.areaName] : undefined,
+                            opponent: String = settings.opponent.category;
+
+                        if (!grassArea || opponent !== "Wild") {
+                            return;
+                        }
+
+                        grassArea.pokemonEncountered = true;
+                    },
+                    "onOpenItemsMenu": function (mod: ModAttachr.IModAttachrMod, eventName: string, items: any[]): void {
+                        let grassMap: IMap = this.player.grass ? <IMap>this.AreaSpawner.getMap(this.player.grass.mapName) : undefined,
+                            grassArea: IArea = grassMap ? <IArea>grassMap.areas[this.player.grass.areaName] : undefined;
+
+                        if (!this.BattleMover.getInBattle() || !(grassArea && grassArea.pokemonEncountered)) {
+                            return;
+                        }
+
+                        for (let i: number = items.length - 1; i > -1; i -= 1) {
+                            let currentItem: IItemSchema = this.MathDecider.getConstant("items")[items[i].item];
+                            if (currentItem.category === "PokeBall") {
+                                items.splice(i, 1);
```

Comment:
It iterates from the end of the array so that it can make sure to get both of them.